apps/vaporgui/VolumeSubtabs.cpp

```diff
@@ -15,163 +24,29 @@ void VolumeVariablesSubtab::Update(DataMgr *dataMgr, ParamsMgr *paramsMgr, Rende
 
 VolumeAppearanceSubtab::VolumeAppearanceSubtab(QWidget* parent) {
     setupUi(this);
-    verticalLayout->insertWidget(0, _tfe = new TFEditorVolume);
-    
-    _densitySlider = new QSliderEdit();
-    _densitySlider->SetLabel("Volume Density");
-    _densitySlider->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Minimum);
-    connect(_densitySlider, SIGNAL(valueChanged(double)), this, SLOT(_densitySlider_valueChanged(double)));
-    _raytracingFrame->layout()->addWidget(_densitySlider);
-    
-    _params = nullptr;
-    
-    _samplingRateComboBox->blockSignals(true);
-    vector<float> samplingOptions = VolumeParams::GetSamplingRateMultiples();
-    for (double rate : samplingOptions)
-        _samplingRateComboBox->addItem(GetQStringForSamplingRate(rate));
-    _samplingRateComboBox->blockSignals(false);
     
-    // Set up lighting parameter widgets
-    _ambientWidget->SetLabel( QString("Ambient") );
-    _ambientWidget->SetDecimals( 2 );
-    _ambientWidget->SetExtents( 0.0, 1.0 );
-    _ambientWidget->SetIntType( false );
+    verticalLayout->insertWidget(0, _pg = new PGroup);
+    _pg->Add((new PTFEditor(RenderParams::_variableNameTag, {PTFEditor::Default}, "Transfer Function"))->ShowColormapBasedOnParam(VolumeParams::UseColormapVariableTag, false));
+    _pg->Add((new PTFEditor(RenderParams::_colorMapVariableNameTag, {PTFEditor::Histogram, PTFEditor::Colormap}, "Colormap Transfer Function"))->ShowBasedOnParam(VolumeParams::UseColormapVariableTag));
     
-    _diffuseWidget->SetLabel( QString("Diffuse") );
-    _diffuseWidget->SetDecimals( 2 );
-    _diffuseWidget->SetExtents( 0.0, 1.0 );
-    _diffuseWidget->SetIntType( false );
+    PSection *rtp = new PSection("Volume Parameters");
+    _pg->Add(rtp);
+    rtp->Add(new PStringDropdownHLI<VolumeParams>("Raytracing Algorithm", VolumeParams::GetAlgorithmNames(VolumeParams::Type::DVR), &VolumeParams::GetAlgorithm, &VolumeParams::SetAlgorithmByUser));
+    rtp->Add(new PEnumDropdown(VolumeParams::SamplingRateMultiplierTag, {"1x", "2x", "4x", "8x", "16x"}, {1, 2, 4, 8, 16}, "Sampling Rate Multiplier"));
+    rtp->Add((new PDoubleSliderEdit(VolumeParams::VolumeDensityTag, "Volume Density"))->EnableDynamicUpdate());
```

Comment:
Do PWidgets support "tool tips"? I have no idea what Volume Density does and it would sure be helpful to have a tool tip.

Reply:
Yes and done. By the way the slider dynamically shows the change so a user should be able to figure it out pretty quick by messing with it.
![out](https://user-images.githubusercontent.com/2772687/78306216-11f5e980-7500-11ea-812b-2d253ed7c0fc.gif)